File: matrix.java
import java.util.Scanner;

public class MatrixMultiplication {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter number of rows and columns of the first matrix:");
        int r1 = scanner.nextInt();
        int c1 = scanner.nextInt();

        System.out.println("Enter number of rows and columns of the second matrix:");
        int r2 = scanner.nextInt();
        int c2 = scanner.nextInt();

        if (c1 != r2) {
            System.out.println("Multiplication of matrices is not possible. First matrix's columns must equal second matrix's rows.");
            return;
        }

        int[][] m1 = new int[r1][c1];
        int[][] m2 = new int[r2][c2];
        int[][] result = new int[r1][c2];

        System.out.println("Enter elements of the first matrix:");
        for (int i = 0; i < r1; i++) {
            for (int j = 0; j < c1; j++) {
                m1[i][j] = scanner.nextInt();
            }
        }

        System.out.println("Enter elements of the second matrix:");
        for (int i = 0; i < r2; i++) {
            for (int j = 0; j < c2; j++) {
                m2[i][j] = scanner.nextInt();
            }
        }

        // Initialize the result matrix with zeros
        for (int i = 0; i < r1; i++) {
            for (int j = 0; j < c2; j++) {
                result[i][j] = 0;
            }
        }

        // Matrix multiplication logic
        for (int i = 0; i < r1; i++) {
            for (int j = 0; j < c2; j++) {
                for (int k = 0; k < c1; k++) {
                    result[i][j] += m1[i][k] * m2[k][j];
                }
            }
        }

        System.out.println("Multiplied matrix:");
        for (int i = 0; i < r1; i++) {
            for (int j = 0; j < c2; j++) {
                System.out.print(result[i][j] + " ");
            }
            System.out.println();
        }

        scanner.close();
    }
}

File: matrix.c
#include<stdio.h>
#include<math.h>
int main()
{
int r1,c1,r2,c2;
printf("enter number of rows and columns of first matrix:\n");
scanf("%d%d", &r1,&c1);

printf("enter number of rows and columns of second matrix:\n");
scanf("%d%d", &r2,&c2);
if(c1!=r2){
printf("multiplication of matrices is not possible.First column equals to second row:");

}
int m1[r1][c1],m2[r2][c2],result[r1][c2],i,j,k;
printf("enter elements of first matrix:\n");
for(i=0;i<r1;i++){
for(j=0;j<c1;j++){
scanf("%d", &m1[i][j]);
}
}
printf("enter elements of second matrix:\n");
for(i=0;i<r2;i++){
for(j=0;j<c2;j++){
scanf("%d", &m2[i][j]);
}
}
for(i=0;i<r1;i++){
for(j=0;j<c2;j++){
result[i][j]=0;
}
}
for(i=0;i<r1;i++){
for(j=0;j<c2;j++){
for(k=0;k<c1;k++){
result[i][j]=m1[i][k]*m2[k][j];
}
}
}
printf("multiplied matrix:\n");
for(i=0;i<r1;i++){
for(j=0;j<c2;j++){
printf("%d", result[i][j]);
}
printf("\n");
}
return 0;
}





